src/lib/fcitx/instance.cpp

```diff
@@ -1684,10 +1684,10 @@ std::string Instance::inputMethodIcon(InputContext *ic) {
     const auto *entry = inputMethodEntry(ic);
     auto *engine = inputMethodEngine(ic);
 
-    if (engine) {
+    if (engine && entry) {
```

Comment:
This never happens, because valid engine ensures valid entry.

Reply:
done